### ISSUE TITLE
Shotcuts using the Meta key in Windows do not work

### ISSUE THREAD

**Issue by jnsebgosselin** [spyder-ide/spyder#7853]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

As mentionned in https://github.com/spyder-ide/spyder/pull/7768#issuecomment-416251033, using the `Meta` or `Windows` key is not a valid key sequence for shortcuts  in Windows 10.

![image](https://user-images.githubusercontent.com/10170372/45236188-2aad0780-b2a9-11e8-91db-d14c479de89e.png)

The event associated with the key sequence is intercepted by Windows and is never sent to Spyder. So to my knowledge, there is not much we can do about this...

Unfortunately, in Spyder 14 shortcuts use the `Meta` key by default in their key sequence. This means that, 14 shortcuts are not usable by default in Spyder.

I think that we need to re-define other key sequences for these shortcuts and avoid using the `Meta` key at all cost.

![image](https://user-images.githubusercontent.com/10170372/45236490-f9810700-b2a9-11e8-960e-4d9e8f79d436.png)

**Comment by jnsebgosselin** [comment]
This issue is also discussed in https://github.com/spyder-ide/spyder/issues/7146.

**Comment by ccordoba12** [comment]
> I think that we need to re-define other key sequences for these shortcuts and avoid using the Meta key at all cost.

Agreed. Please try to find suitable shortcuts for them but avoid using `Ctrl+Shift+<letter>` because we try to reserve those to switch among plugins.

**Comment by jnsebgosselin** [comment]
So the `Next word` and `Previous word` shortcut were fixed in PR #7874, so now we are down to 12 problematic shortcuts using the ` Meta` key.

![image](https://user-images.githubusercontent.com/10170372/45391568-76391b80-b5f1-11e8-9f05-ff40d8255b48.png)

**Comment by jnsebgosselin** [comment]
@ccordoba12 regarding the  `start of line` and `end of line`  shortcuts.

We have in the Editor actions hard-coded for the <kbd>Home</kbd> and <kbd>End</kbd> keys [here](https://github.com/spyder-ide/spyder/blob/master/spyder/plugins/editor/widgets/codeeditor.py#L2935). This was implemented before the `start of line` and `end of line`  shortcuts were added to our keyboard shortcut preferences.

The reason for this was to provide consistency in macOS (see Issue #495), because <kbd>Home</kbd> and <kbd>End</kbd> do not corresponds to builtin key seq for the  `start of line` and `end of line`  shortcut in this OS, as you can see in the table below. Moreover, this also provided the opportunity to implement a smart HOME feature .

> Smart HOME feature: cursor is first moved at indentation position, then at the start of the line.

![image](https://user-images.githubusercontent.com/10170372/45391758-476f7500-b5f2-11e8-90bf-133e08a507c6.png)

**My question is:**

Should we now remove these hard-coded actions and define <kbd>Home</kbd> and <kbd>End</kbd> as the default key seq for the  `start of line` and `end of line` shortcuts in our preferences? This would remove duplication, while fixing the current `Meta` key issue for these 2 shortcuts.

**Comment by ccordoba12** [comment]
> Should we now remove these hard-coded actions and define Home and End as the default key seq for the start of line and end of line shortcuts in our preferences?

Ok, please go ahead with it.

**Comment by jnsebgosselin** [comment]
> Should we now remove these hard-coded actions and define Home and End as the default key seq for the start of line and end of line shortcuts in our preferences?

>> Ok, please go ahead with it.

Sorry for the confusion @ccordoba12. I've decided to make this issue an `Epic` and converted the comment you've replied to to Issue #7884. I should not have deleted the comment here though... my mistake.

**Comment by ccordoba12** [comment]
Ok, no problem.

**Comment by zkoynz** [comment]
👍 progress? To be completely honest, the only reason I want this addressed is for quick access to 'Shortcuts Summary' 😶 

Thanks for your hard work!

**Comment by jnsebgosselin** [comment]
> 
> 
> 👍 progress? To be completely honest, the only reason I want this addressed is for quick access to 'Shortcuts Summary' 😶
> 
> Thanks for your hard work!

This shortcut is hardcoded in `spyder.app.mainwindow.py`, so we need to change that there. This can be done for Spyder4. I still need to better understand how shortcuts work in macOS before I change this though.
